_docs/install/kubernetes-csi/dynamic-provisioning.md

```diff
@@ -0,0 +1,281 @@
+---
+layout: guide
+title: StorageOS Docs - Kubernetes (CSI)
+anchor: install
+module: install/kubernetes-csi/dynamic-provisioning
+---
+
+# Dynamic Provisioning (CSI)
+
+StorageOS supports [Container Storage Interface](https://kubernetes-csi.github.io/docs/) which enables kubernetes to use
+a unix domain socket to communicate with StorageOS driver. Using this setup
+removes the need to explicitly create secrets for kubernetes to talk to
+StorageOS driver.
+
+StorageOS volumes can be created on-demand through dynamic provisioning.
+
+1. Adminstrators create storage classes to define different types of storage.
+1. Users create a persistent volume claim (PVC).
+1. The user references the PVC in a pod.
+
+The [StorageOS Helm chart for CSI deployment](https://github.com/storageos/helm-chart/tree/csi-deployment) includes the
```

Comment:
NB we should have one helm chart that can deploy both CSI and non-CSI, controlled by an option.